cmd/ctr/checkpoint.go

```diff
@@ -67,6 +68,24 @@ var checkpointCommand = cli.Command{
 		if err := json.Unmarshal(info.Spec.Value, &spec); err != nil {
 			return err
 		}
+		stopped := context.Bool("exit")
+		// if the container will still be running after the checkpoint make sure that
+		// we pause the container and give us time to checkpoint the filesystem before
+		// it resumes execution
+		if !stopped {
+			if _, err := containers.Pause(ctx, &execution.PauseRequest{
+				ID: id,
+			}); err != nil {
+				return err
+			}
+			defer func() {
+				if _, err := containers.Resume(ctx, &execution.ResumeRequest{
```

Comment:
Its fine, this tool is no supported and very basic on its implementation.  It does not handle edge cases like this